integration-tests/test_man_page.py

```diff
@@ -0,0 +1,47 @@
+"""
+This Python module contains integration tests for rhc.
+It uses pytest-client-tools Python module. More information about
+this module could be found: https://github.com/ptoscano/pytest-client-tools/
+"""
+
+import subprocess
+import pytest
+
+
+def test_man_page_synopsis():
+    """This test verifies format of man page"""
+    command_op = subprocess.check_output(["man", "rhc"]).decode("utf-8")
+    assert (
+        "rhc [GLOBAL OPTIONS] command [COMMAND OPTIONS] " "[ARGUMENTS...]"
+    ) in command_op
+
+
+def test_man_page_global_options():
+    """Test verifies global option are present in man page"""
+    command_op = subprocess.check_output(["man", "rhc"]).decode("utf-8")
+    assert "--help, -h" in command_op
+    assert "--no-color" in command_op
+    assert "--version, -v" in command_op
+
+
+@pytest.mark.parametrize("command", ["connect", "disconnect", "status", "help"])
+def test_man_page_commands(command):
+    """Test verifies if man page displays existing commands"""
+    command_op = subprocess.check_output(["man", "rhc"]).decode("utf-8")
+    assert command in command_op
+
+
+@pytest.mark.parametrize(
+    "option",
+    [
+        "--activation-key, -a",
+        "--organization, -o",
+        "--password, -p",
+        "--server",
```

Comment:
No, there can not be such option `--server`. It is reason, why your the test is failing.

Reply:
I found this option in man page only
rhc(8)                                                                       System Manager's Manual                                                                      rhc(8)

NAME
       rhc - control the system's connection to Red Hat

SYNOPSIS
       rhc

              [--help|-h]
              [--no-color]
              [--version|-v]

       Usage:

              rhc [GLOBAL OPTIONS] command [COMMAND OPTIONS] [ARGUMENTS...]

GLOBAL OPTIONS
       --help, -h: show help

       --no-color:

       --version, -v: print the version

COMMANDS
connect
       Connects the system to Red Hat

       --activation-key, -a="": register with KEY

       --organization, -o="": register with ID

       --password, -p="": register with PASSWORD

       **--server="": register against URL**

       --username, -u="": register with USERNAME

Reply:
and also tests are passing downstream:
                out: integration-tests/test_man_page.py::test_man_page_synopsis PASSED        [ 54%]
                out: integration-tests/test_man_page.py::test_man_page_global_options PASSED  [ 58%]
                out: integration-tests/test_man_page.py::test_man_page_commands[connect] PASSED [ 62%]
                out: integration-tests/test_man_page.py::test_man_page_commands[disconnect] PASSED [ 66%]
                out: integration-tests/test_man_page.py::test_man_page_commands[status] PASSED [ 70%]
                out: integration-tests/test_man_page.py::test_man_page_commands[help] PASSED  [ 75%]
                out: integration-tests/test_man_page.py::test_man_page_connect_options[--activation-key, -a] PASSED [ 79%]
                out: integration-tests/test_man_page.py::test_man_page_connect_options[--organization, -o] PASSED [ 83%]
                out: integration-tests/test_man_page.py::test_man_page_connect_options[--password, -p] PASSED [ 87%]
                out: integration-tests/test_man_page.py::test_man_page_connect_options[--server] PASSED [ 91%]
                out: integration-tests/test_man_page.py::test_man_page_connect_options[--username, -u] PASSED [ 95%]
                out: integration-tests/test_version.py::test_version PASSED

Reply:
This CLI option was marked as hidden: https://github.com/RedHatInsights/rhc/commit/15af69150276036c7f7dacb984995ae42abf73ea

This CLI opion will be completely removed in the future.

The man page is generated from Go source code.

Maybe, we should test that this CLI option is NOT present in man page, because this CLI option was just wrong idea.

I'm little bit confused. Why and how do you test downstream? This GitHub repository is for upstream.

Reply:
[PR](https://github.com/RedHatInsights/rhc/pull/33)  was made for rhc 0.2.4 then why do we still see that --server option in man page on RHEL 10 build 0.2.4-6.el10.x86_64
if this option is removed then it does make sense to test it otherwise(negative case). I will update the PR.

Reason for testing downstream right now is that we are not able to run all tests upstream due to lack of candlepin/insights credentials  accessibility upstream. We have a pending task to fix this issue.

Reply:
Archana, please just remove this test for presence of `--server` option. This CLI option should never been introduced. It just cannot be used by anyone in production. Thanks.